libs/feature/search/src/lib/results-table/results-table.component.spec.ts

```diff
@@ -0,0 +1,209 @@
+  })
+
+  describe('get a list of formats and sorts them depending on priority', () => {
+    it('returns a list of unique formats', () => {
+      expect(component.getRecordFormats(DATASET_RECORDS[0])).toEqual([
+        'geojson',
+        'shp',
+        'pdf',
+      ])
+    })
+  })
+  describe('get the badge color for given format', () => {
+    it('returns the color for its format', () => {
+      expect(
+        component.getBadgeColor(
+          component.getRecordFormats(DATASET_RECORDS[0])[0]
+        )
+      ).toEqual('#1e5180') // geojson
+    })
+  })
+
+  describe('sorting', () => {
+    describe('#setSortBy', () => {
+      it('calls the facade to change sort order', () => {
+        component.setSortBy('title', 'asc')
+        expect(searchService.setSortBy).toHaveBeenCalledWith(['asc', 'title'])
+      })
+    })
+    describe('#isSortedBy', () => {
+      it('returns null if not sorted by this column', async () => {
+        searchFacade.sortBy$.next(['desc', 'owner'])
+        const sort = await firstValueFrom(component.isSortedBy('title'))
+        expect(sort).toBe(null)
+      })
+      it('returns the sort order if the current sortBy is for this column', async () => {
+        searchFacade.sortBy$.next(['desc', 'title'])
+        const sort = await firstValueFrom(component.isSortedBy('title'))
+        expect(sort).toBe('desc')
+      })
+      it('returns true if the current sortBy is for this column (multiple sorts)', async () => {
+        searchFacade.sortBy$.next([
+          ['asc', 'score'],
+          ['desc', 'title'],
+        ])
+        expect(await firstValueFrom(component.isSortedBy('title'))).toBe('desc')
+        expect(await firstValueFrom(component.isSortedBy('score'))).toBe('asc')
+        expect(await firstValueFrom(component.isSortedBy('owner'))).toBe(null)
+      })
+    })
+  })
+
+  describe('selection', () => {
+    beforeEach(() => {
+      searchFacade.results$.next([
+        {
+          uniqueIdentifier: '1',
+        },
+        {
+          uniqueIdentifier: '2',
+        },
+        {
+          uniqueIdentifier: '3',
+        },
+      ] as any)
+    })
+
+    describe('#isChecked', () => {
+      it('should return true when the record is in the selectedRecords array', async () => {
+        selectionService.selectedRecordsIdentifiers$.next(['1', '2'])
+        const record = { uniqueIdentifier: '2' } as CatalogRecord
+        expect(await firstValueFrom(component.isChecked(record))).toBe(true)
+      })
+
+      it('should return false when the record is not in the selectedRecords array', async () => {
+        selectionService.selectedRecordsIdentifiers$.next(['1', '2', '3'])
+        const record = { uniqueIdentifier: '4' } as CatalogRecord
+        expect(await firstValueFrom(component.isChecked(record))).toBe(false)
+      })
+    })
+
+    describe('#handleRecordSelectedChange', () => {
+      it('should call selectRecords when checkbox is clicked', () => {
```

Comment:
Possible misleading comment, as the click is not part of the test, if at some point the event is changed to another, the comment might not be changed.